config/seed-peer-generator/src/main.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+
+use libra_config::config::{PersistableConfig, HANDSHAKE_VERSION};
+use libra_network_address::NetworkAddress;
+use libra_secure_json_rpc::JsonRpcClient;
+use libra_types::{on_chain_config::ValidatorSet, validator_info::ValidatorInfo, PeerId};
+use std::{collections::HashMap, convert::TryFrom, path::PathBuf};
+use structopt::StructOpt;
+use libra_types::account_address::AccountAddress;
+
+// TODO: Use the definition from network?
+pub type SeedPeersConfig = HashMap<PeerId, Vec<NetworkAddress>>;
+
+#[allow(clippy::large_enum_variant)]
+#[derive(Debug, StructOpt)]
+#[structopt(about = "Tool to generate configs from chain")]
+enum Args {
+    #[structopt(about = "Generate seed peers config")]
+    GenerateSeedPeers(GenerateArgs),
+}
+
+#[derive(Debug, StructOpt)]
+struct GenerateArgs {
+    #[structopt(short = "o", long, parse(from_os_str))]
+    /// The output directory
+    output_dir: PathBuf,
+    #[structopt(short = "e", long)]
+    /// JSON RPC endpoint
+    endpoint: String,
+    #[structopt(short = "a", long)]
+    /// Specify the AccountAddress to retrieve peers for
+    account_address: AccountAddress,
+}
+
+fn main() {
+    let args = Args::from_args();
+
+    match args {
+        Args::GenerateSeedPeers(args) => generate_seed_peers_config(args)
+    };
+}
+
+fn generate_seed_peers_config(args: GenerateArgs) {
+    let mut seed_peers_config = SeedPeersConfig::new();
+    let validator_set = get_validator_set(args.endpoint, args.peer_id).unwrap();
+
+    // Add the validators to the config!
+    if let Some(validator_set) = validator_set {
+        for validator in validator_set.payload().iter() {
+            let (peer_id, seed_addrs) = to_seed_peer(validator);
+            seed_peers_config.insert(peer_id, seed_addrs);
+        }
+    } else {
+        panic!("No ValidatorSet found")
+    }
+
+    // Save to a file for loading later
+    seed_peers_config
+        .save_config(args.output_dir.join("seed_peers.yaml"))
+        .expect("Unable to save seed peers config");
+}
+
+/// Retrieve validator set from the JSON-RPC endpoint
+fn get_validator_set(endpoint: String, peer_id: PeerId) -> anyhow::Result<Option<ValidatorSet>> {
+    let json_rpc = JsonRpcClient::new(endpoint);
+    let account_state = json_rpc.get_account_state(peer_id, None)?;
+    Ok(account_state.get_validator_set()?)
```

Comment:
just for fun, you could also use `.map_err(|e| e.into())`
I don't know which one is better...